src/Storages/MergeTree/MergeTreeIndexVectorSimilarity.cpp

```
@@ -140,6 +143,8 @@ void USearchIndexWithSerialization::deserialize(ReadBuffer & istr)
 {
     auto callback = [&istr](void * from, size_t n)
     {
+        /// USearch may call callback from noexcept function
+        LockMemoryExceptionInThread lock_memory_exceptions;
```

Shouldn't we return false in case of exception here?

---

I thought about this but in this case we will got proper exception, hence decided to suppress them, anyway allocations should not be big, on CI it failed due to fault injection.

---

> this case we will got proper exception

Did you mean, `will _NOT_`? - afaiu, if false is returned here, then `ErrorCodes::INCORRECT_DATA` will be thrown

In general, is it possible to disable fault injection in a scope? Then we'd simply disable it here. It looks as a proper way to do it.

Also, if we suppress the exceptions in the callback, - deserialization can still fail since it can try to use uninitialized variables as size to allocate some memory, - see example here - https://github.com/ClickHouse/usearch/blob/50cb56caa1ef19317717f85043ec8c1e56712569/include/usearch/index_dense.hpp#L1117. 
`maxtix_cols` will be not initialized if we will throw exception and suppress it here, https://github.com/ClickHouse/usearch/blob/50cb56caa1ef19317717f85043ec8c1e56712569/include/usearch/index_dense.hpp#L1107-L1110

---

Actually I guess the allocation may be not small, so better to avoid allowing memory overcommit here and fail instead. Done.